libs/vre/shared/app-resource-page/src/lib/resource-representation.component.ts

```diff
@@ -15,25 +15,23 @@ import { filter, map } from 'rxjs/operators';
 
 @Component({
   selector: 'app-resource-representation',
-  template: ` <div class="representation-container center" [ngSwitch]="representationToDisplay.fileValue.type">
-    <app-still-image
-      #stillImage
-      class="dsp-representation stillimage"
-      [attachedProject]="attachedProject$ | async"
-      *ngSwitchCase="representationConstants.stillImage"
-      [resource]="resource.res" />
-    <app-still-image
-      #stillImage
-      class="dsp-representation stillimage"
-      *ngSwitchCase="representationConstants.externalStillImage"
-      [attachedProject]="attachedProject$ | async"
-      [resource]="resource.res" />
-
+  template: ` <div class="representation-container center" [ngSwitch]="true">
+    <ng-container *ngIf="attachedProject$ | async as attachedProject">
+      <app-still-image
+        #stillImage
+        class="dsp-representation stillimage"
+        *ngSwitchCase="
+          representationToDisplay.fileValue.type === representationConstants.stillImage ||
+          representationToDisplay.fileValue.type === representationConstants.externalStillImage
+        "
+        [attachedProject]="attachedProject"
+        [resource]="resource.res" />
+    </ng-container>
     <app-document
       #document
       class="dsp-representation document"
       [class.pdf]="representationToDisplay.fileValue.filename.split('.').pop() === 'pdf'"
-      *ngSwitchCase="representationConstants.document"
+      *ngSwitchCase="representationToDisplay.fileValue.type === representationConstants.document"
```

Comment:
*ngSwitchCase should represent a value, and not a condition. Why is it like this ?

Reply:
@derschnee68 stupid angular templates do not support swicthCase multiple or operators, probably that is why Dominique had multiple still image component instances.

Reply:
How about using IF operator inside switch case?
`representationToDisplay.fileValue.type === XXX || representationToDisplay.fileValue.type === YYY`

Reply:
@mpro7 That seems much better, and we keep the original switchCase.
It looked all so good, let's not spoil it because it does not work as expected. There must be some other issues

Reply:
@mpro7 this is exactly how it is done now: 
```
*ngSwitchCase="
          representationToDisplay.fileValue.type === representationConstants.stillImage ||
          representationToDisplay.fileValue.type === representationConstants.externalStillImage
        "
```

Reply:
I don't see it in the file in one `ngSwitchCase` like in above statement. The other option would be to use `ngSwitchDefault` or `ngTemplateOutlet` directives.

Reply:
@mpro7 I don't know what you are proposing, could you please add your working copy of this file in this pr?

Reply:
Please ignore my comment, I misunderstood the first question in this thread. I can see now lines 23-26 where it is used as as it should be.